scripts/gen_regex.py

```diff
@@ -4,89 +4,74 @@
 from pkg_resources import resource_filename
 
 
+CATEGORIES = [unicodedata.category(chr(i)) for i in range(0x110000)]
 DATA_PATH = pathlib.Path(resource_filename('wordfreq', 'data'))
 
 
+def func_to_regex(func):
+    """
+    Given a function that returns True or False for a numerical codepoint,
+    return a regex character class accepting the characters resulting in True.
+    Ranges separated only by unassigned characters are merged for efficiency.
```

Comment:
Aside from the code being shorter (since we know I have an irrational need to make code shorter), the main reason for this rewrite is the reduction in accepted unassigned characters.  This feels vaguely cleaner to me, but I'm not 100% sure.  Probably at least for the emoji and combining mark regexes the default for unknown characters should be "no".